app/models/service.rb

```diff
@@ -528,6 +529,11 @@ def archive_as_deleted
     ::DeletedObject.create!(object: self, owner: account)
   end
 
+  def destroy_related_backend_apis
+    return if provider_can_use?(:api_as_product)
+    backend_apis.with_single_service.destroy_all
```

Comment:
No need to look for `with_single_service`

If it is under `api_as_product` then we do not destroy it, otherwise we just destroy the only one backend_api

```suggestion
    backend_api.&.destroy
```